cmd/clusterctl/client/cluster/installer.go

```diff
@@ -90,6 +109,58 @@ func installComponentsAndUpdateInventory(components repository.Components, provi
 	return providerInventory.Create(inventoryObject)
 }
 
+// waitForReady waits till the installed components are ready.
+func (i *providerInstaller) waitForReady() error {
+	log := logf.Log
+	log.Info("Waiting for providers to be available...")
+
+	if err := i.waitDeploymentsReady(); err != nil {
+		return err
+	}
+
+	// check for services too?
+
+	return nil
+}
+
+// waitDeploymentsReady waits till the installed deployments are ready.
+func (i *providerInstaller) waitDeploymentsReady() error {
+	for _, components := range i.installQueue {
+		for _, obj := range components.Objs() {
+			if obj.GroupVersionKind().Kind == "Deployment" {
```

Comment:
According to https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#controllers--watching-namespace we can eventually check for the container named manager in order to better identify the controller.